pkg/controller/migmigration/annotation.go

```diff
@@ -477,6 +487,47 @@ func (t *Task) deleteNamespaceLabels(client k8sclient.Client, namespaceList []st
 	return nil
 }
 
+// deleteLabels will delete all migration.openshift.io/migrated-by labels
+// from the application upon successful completion
+func (t *Task) deleteLabels() error {
+	client, GVRs, err := t.getResourcesForDelete()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+
+	for _, gvr := range GVRs {
+		for _, ns := range t.destinationNamespaces() {
```

Comment:
Having an addition listing per-namespace is explained by the issue I saw, by listing resources in cluster scope with
```go
client.Resource(gvr).List(*listOptions) 
```

Such untyped client had issues picking up CRs with label selectors across multiple namespaces, and instead returned a full list of resources across cluster.